header-rewrite-filter/header_processor.h

```diff
@@ -91,6 +91,24 @@ class SetHeaderProcessor : public HeaderProcessor {
   std::vector<std::string> header_vals_; // header values to set
 };
 
+class AppendHeaderProcessor : public HeaderProcessor {
+public:
+  AppendHeaderProcessor() {}
+  virtual ~AppendHeaderProcessor() {}
+  virtual absl::Status parseOperation(std::vector<absl::string_view>& operation_expression, std::vector<absl::string_view>::iterator start);
+  virtual absl::Status executeOperation(Http::RequestOrResponseHeaderMap& headers, SetBoolProcessorMapSharedPtr bool_processors);
+private:
+  // Note: the values returned by these functions must not outlive the AppendHeaderProcessor object
+  const std::string& getKey() const { return header_key_; }
```

Comment:
Nit: Since Envoy and this extension use `string_view`, this would be more convenient if it returned `string_view` rather than `string&`. A `string_view` is really just a slightly-more-convenient and slightly-faster-in-some-cases version of `string&`. It'd also make the filter more consistent, to use `string_view` everywhere. But not a huge deal.